index.bs

```diff
@@ -1985,6 +1989,124 @@ The following extended attributes are applicable to operation arguments:
+*   a [=string type=],
+*   a [=nullable type=] whose [=inner type=] is a [=JSON value=],
+*   a [=union type=] where all of its [=member types=] are [=JSON values=],
+*   a [=sequence type=] whose parameterized type is a [=JSON value=],
+*   a [=dictionary=] where all of its [=dictionary members|members=] are [=JSON values=],
+*   a [=record=] where all of its [=map/values=] are [=JSON values=], and
+*   an [=object type=],
+*   an [=interface type=] that has a “toJSON” method declared on itself or
+    one of its [=inherited interface|inherited=] or
+    [=consequential interface|consequential=] interface.
+
+How the “toJSON” [=regular operation=]
+is made available on an object in a language binding,
+and how exactly the [=JSON values=] are converted into a JSON string,
+is language binding specific.
+
+Note: In the ECMAScript language binding,
+this is done by exposing a “toJSON” method
+which returns the [=JSON value=]
+converted into an ECMAScript value
+that can be turned into a JSON string
+by the <code>[=JSON.stringify=]</code> function.
+Additionaly, in the ECMAScript language binding,
+the “toJSON” operation can take a [{{Default}}] [=extended attribute=],
+in which case the [=default toJSON operation=] is exposed instead.
+
+<div class="example">
+
+    The following [=IDL fragment=] defines an interface <code class="idl">Transaction</code>
+    that has a “toJSON” method defined in prose,
+    and an interface <code class="idl">Account</code>,
+    which relies on the [=default toJSON operation=]
+    (its “toJSON” operation is declared with a [{{Default}}] [=extended attribute=]):
+
+    <pre highlight="webidl">
+        interface Transaction {
+          readonly attribute Account from;
+          readonly attribute Account to;
+          readonly attribute double amount;
+          readonly attribute DOMString description;
+          readonly attribute unsigned long number;
+          TransactionJSONValue toJSON();
+        };
+
+        dictionary TransactionJSONValue {
+          Account from;
+          Account to;
+          double amount;
+          DOMString description;
+        };
+
+        interface Account {
+          attribute DOMString name;
+          attribute unsigned long number;
+          [Default] any toJSON();
+        };
+    </pre>
+
+    Assuming each of the attributes of <code class="idl">Transaction</code> is backed by an associated value,
+    so that each [=attribute getter=] returns the corresponding associated value,
+    then the “toJSON” [=regular operation=] could be defined as follows:
+
+    <blockquote>
+        <div algorithm="example tojson">
+
+            The “toJSON” operation of the <code class="idl">Transaction</code> interface
+            must follow these steps:
+
+            1.  Let |json| be a new <code class="idl">TransactionJSONValue</code> dictionary.
+            1.  Set |json|[“from”] to this <code class="idl">Transaction</code>'s associated "from" value.
```

Comment:
Yeah, the spec seems to really like using curly quotes for operation names, but for dictionary fields at least I assumed we were following the infra-style dictionaries-as-ordered-maps-with-string-keys and infra-strings don't use curly quotes.

In the future maybe it'd be cool to use `<code>` for operation names instead of curly quotes. *shrug*